src/ComputeSharp.D2D1/Shaders/Interop/D2D1ResourceTextureManager.cs

```diff
@@ -0,0 +1,487 @@
+    /// <param name="channelDepth">The channel depth for the resource to create.</param>
+    /// <param name="filter">The filter for the resource to create.</param>
+    /// <param name="extendModes">The extend modes for the resource to create.</param>
+    /// <param name="data">The data to load in the resource to create.</param>
+    /// <param name="strides">The strides for the data supplied for the resource to create.</param>
+    /// <exception cref="Exception">Thrown if creating or initializing the <see cref="D2D1ResourceTextureManager"/> instance failed.</exception>
+    public D2D1ResourceTextureManager(
+        Guid resourceId,
+        ReadOnlySpan<uint> extents,
+        D2D1BufferPrecision bufferPrecision,
+        D2D1ChannelDepth channelDepth,
+        D2D1Filter filter,
+        ReadOnlySpan<D2D1ExtendMode> extendModes,
+        ReadOnlySpan<byte> data,
+        ReadOnlySpan<uint> strides)
+        : this(
+              &resourceId,
+              extents,
+              bufferPrecision,
+              channelDepth,
+              filter,
+              extendModes,
+              data,
+              strides)
+    {
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="D2D1ResourceTextureManager"/> instance with the specified parameters.
+    /// </summary>
+    /// <param name="resourceId">The resource id for the resource to create.</param>
+    /// <param name="extents">The extents of the resource to create.</param>
+    /// <param name="bufferPrecision">The buffer precision for the resource to create.</param>
+    /// <param name="channelDepth">The channel depth for the resource to create.</param>
+    /// <param name="filter">The filter for the resource to create.</param>
+    /// <param name="extendModes">The extend modes for the resource to create.</param>
+    /// <param name="data">The data to load in the resource to create.</param>
+    /// <param name="strides">The strides for the data supplied for the resource to create.</param>
+    /// <exception cref="Exception">Thrown if creating or initializing the <see cref="D2D1ResourceTextureManager"/> instance failed.</exception>
+    private D2D1ResourceTextureManager(
+        Guid* resourceId,
+        ReadOnlySpan<uint> extents,
+        D2D1BufferPrecision bufferPrecision,
+        D2D1ChannelDepth channelDepth,
+        D2D1Filter filter,
+        ReadOnlySpan<D2D1ExtendMode> extendModes,
+        ReadOnlySpan<byte> data,
+        ReadOnlySpan<uint> strides)
+    {
+        // Manually validate the parameters here to ensure the native objects wouldn't try
+        // to read out of bounds and potentially AV. For those cases, we just throw here.
+        if (extents.Length == 0 ||
+            extendModes.Length == 0 ||
+            extents.Length != extendModes.Length ||
+            (data.Length > 0 && strides.Length != extents.Length - 1))
+        {
+            GC.SuppressFinalize(this);
```

Comment:
why is this necessary here? 🤔

Reply:
We're throwing right after, meaning we'll never create the COM object, so there's nothing to release. Adding the finalizer to the finalizer queue is just not needed here, since there's left to do for this object, it's just dead from the start.